### PR TITLE
`mir_array_value`

### PR DESCRIPTION
**Pull request by RyanGlScott**
This adds support for a `mir_array_value` function in the MIR backend, which allows constructing array `SetupValue`s. This largely behaves like `llvm_array_value`s in the LLVM backend, but with the following differences:

* Unlike in the LLVM backend, MIR arrays can have length 0. To account for this possibility, `mir_array_value` has a `MIRType` argument to ensure that SAW can always infer the type of the array, even if there are no element values from which to infer the type.

  Similarly, the SAW remote API's `array()` function now has an optional `element_type` kwarg that can be used to specify the element type in the event that there are no element values.
* Because only the MIR backend makes use of the element type, this is encoded in the `XSetupArray` extension field.

This checks off one box in https://github.com/GaloisInc/saw-script/issues/1859.

### CODE REVIEW

File: src/SAWScript/Crucible/MIR/Override.hs [hunk 1]
```diff
@@ -301,23 +301,37 @@ matchArg opts sc cc cs prepost md actual expectedTy expected@(MS.SetupTerm expec
...
+          [ matchArg opts sc cc cs prepost md (MIRVal elemShp x) y z
+          | (x, z) <- zip (V.toList xs') zs ]
+
+      | Mir.MirVector_PartialVector xs' <- xs
+      , V.length xs' == length zs ->
+        do xs'' <- liftIO $
+             traverse (readMaybeType sym "vector element" (shapeType elemShp)) xs'
+           sequence_
+             [ matchArg opts sc cc cs prepost md (MIRVal elemShp x) y z
+             | (x, z) <- zip (V.toList xs'') zs ]
```
Comment (RyanGlScott) [inline src/SAWScript/Crucible/MIR/Override.hs:325]: Here is how SAW matches array arguments in specifications. Note that `MirVector_Array` (for symbolic, SMT-based arrays) aren't supported.

File: src/SAWScript/Crucible/MIR/ResolveSetupValue.hs [hunk 2]
```diff
@@ -120,11 +120,12 @@ typeOfSetupValue _mcc env _nameEnv val =
             Right mirTy -> return mirTy
         TypedTermSchema s -> X.throwM (MIRPolymorphicType s)
         tp -> X.throwM (MIRInvalidTypedTerm tp)
+    MS.SetupArray elemTy vs ->
+      pure $ Mir.TyArray elemTy (length vs)
```
Comment (RyanGlScott) [inline src/SAWScript/Crucible/MIR/ResolveSetupValue.hs:124]: Here is where the type of `mir_array_value`s are computed.

File: doc/manual/manual.md [hunk 3]
```diff
@@ -2615,6 +2615,13 @@ the value of an array element.
 * `jvm_field_is : JVMValue -> String -> JVMValue -> JVMSetup ()`
 specifies the name of an object field.
 
+In the experimental MIR verification implementation, the following functions
+construct compound values:
+
+* `mir_array_value : MIRType -> [SetupValue] -> SetupValue` constructs an array
+  of the given type whose elements consist of the given values. Supplying the
+  element type is necessary to support length-0 arrays.
```
Comment (RyanGlScott) [inline doc/manual/manual.md:2623]: Documentation

File: saw-remote-api/CHANGELOG.md [hunk 4]
```diff
@@ -13,6 +13,10 @@
   how SAW's MIR verification support works in general, see the `mir_*` commands
   documented in the [SAW
   manual](https://github.com/GaloisInc/saw-script/blob/master/doc/manual/manual.md).
+* The API for `"array"` `setup value`s now has a `"element type"` field. For
```
Comment (bboston7) [inline saw-remote-api/CHANGELOG.md:16]:
```suggestion
* The API for `"array"` `setup value`s now has an `"element type"` field. For
```

**Review by bboston7 — APPROVED** [review]
Looks good! I only looked at the most recent commit because the rest seemed covered by #1919. Let me know if that's not true and I'll go back and review some of the other commits.

**Pull request merged**